src/rpc/client.cpp

```diff
@@ -325,69 +389,104 @@ static UniValue Parse(std::string_view raw)
+        });
+
+        return it == std::end(vRPCConvertParams) ? nullptr : &*it;
     }
 
-    /** Return arg_value as UniValue, and first parse it if it is a non-string parameter */
-    UniValue ArgToUniValue(std::string_view arg_value, const std::string& method, const std::string& param_name)
+    const CRPCConvertParam* FromName(std::string_view method, std::string_view name)
     {
-        return membersByName.count({method, param_name}) > 0 ? Parse(arg_value) : arg_value;
+        auto it = std::ranges::find_if(vRPCConvertParams, [&](const auto& p) {
+            return p.methodName == method && p.paramName == name;
+        });
+
+        return it == std::end(vRPCConvertParams) ? nullptr : &*it;
     }
-};
+} // namespace rpc_convert
 
-CRPCConvertTable::CRPCConvertTable()
+static UniValue ParseParam(const CRPCConvertParam* param, std::string_view raw)
 {
-    for (const auto& cp : vRPCConvertParams) {
-        members.emplace(cp.methodName, cp.paramIdx);
-        membersByName.emplace(cp.methodName, cp.paramName);
-    }
+    // Only parse parameters which have the JSON format; otherwise, treat them as strings.
+    return (param && param->format == ParamFormat::JSON) ? Parse(raw) : UniValue(std::string(raw));
 }
 
-static CRPCConvertTable rpcCvtTable;
 
+/**
+ * Convert command lines arguments to params object when -named is disabled.
+ */
 UniValue RPCConvertValues(const std::string &strMethod, const std::vector<std::string> &strParams)
 {
     UniValue params(UniValue::VARR);
 
-    for (unsigned int idx = 0; idx < strParams.size(); idx++) {
-        std::string_view value{strParams[idx]};
-        params.push_back(rpcCvtTable.ArgToUniValue(value, strMethod, idx));
+    for (std::string_view s : strParams) {
+        params.push_back(ParseParam(rpc_convert::FromPosition(strMethod, static_cast<int>(params.size())), s));
     }
 
     return params;
 }
 
+/**
+ * Convert command line arguments to params object when -named is enabled.
+ *
+ * This accepts arguments in the "PARAM_NAME=PARAM_VALUE" format. Arguments without '=' are
+ * treated as positional. Arguments with '=' are treated as named, except for two special
+ * cases to handle values that contain '=':
+ *
+ *   1. If the PARAM_NAME is not a known parameter name, the code checks if the next
+ *      expected positional parameter is listed as ParamFormat::JSON. If so, and if the entire
+ *      "PARAM_NAME=PARAM_VALUE" string parses as valid JSON, it's treated as positional.
+ *
+ *   2. If the PARAM_NAME is not a known parameter name and the next expected positional parameter
+ *      is listed as a ParamFormat::STRING, the entire "PARAM_NAME=PARAM_VALUE" string is treated
+ *      as a single positional string value.
+ *
+ * For example, the command `bitcoin-cli -named createwallet "my=wallet"`, the
+ * parser initially sees "my=wallet" and attempts to process it as a named
+ * parameter with PARAM_NAME="my". When it finds that "my" is not a valid
```

Comment:
In commit "rpc: Handle -named argument parsing where '=' character is used" (8eafb1e9ee4fff3a9ab4a7dc899bd2d57c29d04f)

Woud suggest replacing `parameter with PARAM_NAME="my".` with `parameter named "my"` so this is shorter and more understandable without the previous context.